### PR TITLE
Update SFrame transform to support BufferSource

### PR DESCRIPTION
**Pull request by youennf**
***
<a href="https://pr-preview.s3.amazonaws.com/youennf/webrtc-insertable-streams/pull/54.html" title="Last updated on Feb 18, 2021, 3:33 PM UTC (01531aa)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-insertable-streams/54/b77c4e1...youennf:01531aa.html" title="Last updated on Feb 18, 2021, 3:33 PM UTC (01531aa)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -252,14 +252,14 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 2. If |frame|.`[[rtcObject]]` is a {{RTCRtpSender}}, set |role| to 'encrypt'.
 3. If |frame|.`[[rtcObject]]` is a {{RTCRtpReceiver}}, set |role| to 'decrypt'.
 4. Let |data| be undefined.
-5. If |frame| is an ArrayBuffer, set |data| to |frame|. // FIXME Support BufferSource
-6. If |frame| is an {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
-7. If |frame| is an {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
+5. If |frame| is a {{BufferSource}}, set |data| to |frame|.
+6. If |frame| is a {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
```
Comment (youennf) [inline index.bs:256]:
```suggestion
6. If |frame| is an {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
```

File: index.bs [hunk 2]
```diff
@@ -252,14 +252,14 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
...
+7. If |frame| is a {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
```
Comment (youennf) [inline index.bs:257]:
```suggestion
7. If |frame| is an {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
```

File: index.bs (hunk 1 above)
Comment (henbos) [inline index.bs]: an RRrrrr

File: index.bs [hunk 3]
```diff
@@ -252,14 +252,14 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
...
 6. If |frame| is an {{RTCEncodedAudioFrame}}, set |data| to |frame|.{{RTCEncodedAudioFrame/data}}
 7. If |frame| is an {{RTCEncodedVideoFrame}}, set |data| to |frame|.{{RTCEncodedVideoFrame/data}}
 8. If |data| is undefined, abort these steps.
-9. Let |buffer| be the result of running the SFrame algorithm with |data| and |role| as parameters. This algorithm is defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a>.
-10. If |frame| is an ArrayBuffer, set |frame| to |buffer|.
-11. If |frame| is an {{RTCEncodedAudioFrame}}, set |frame|.{{RTCEncodedAudioFrame/data}} to |buffer|.
-12. If |frame| is an {{RTCEncodedVideoFrame}}, set |frame|.{{RTCEncodedVideoFrame/data}} to |buffer|.
+9. Let |buffer| be the result of running the SFrame algorithm with |data| and |role| as parameters. This algorithm is defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a> and returns an {{ArrayBuffer}}.
+10. If |frame| is a {{BufferSource}}, set |frame| to |buffer|.
+11. If |frame| is a {{RTCEncodedAudioFrame}}, set |frame|.{{RTCEncodedAudioFrame/data}} to |buffer|.
```
Comment (youennf) [inline index.bs:261]:
```suggestion
11. If |frame| is an {{RTCEncodedAudioFrame}}, set |frame|.{{RTCEncodedAudioFrame/data}} to |buffer|.
```

Comment (youennf) [inline index.bs]:
```suggestion
11. If |frame| is an {{RTCEncodedAudioFrame}}, set |frame|.{{RTCEncodedAudioFrame/data}} to |buffer|.
```

File: index.bs [hunk 4]
```diff
@@ -252,14 +252,14 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
...
+12. If |frame| is a {{RTCEncodedVideoFrame}}, set |frame|.{{RTCEncodedVideoFrame/data}} to |buffer|.
```
Comment (youennf) [inline index.bs:262]:
```suggestion
12. If |frame| is an {{RTCEncodedVideoFrame}}, set |frame|.{{RTCEncodedVideoFrame/data}} to |buffer|.
```

**Comment by youennf** [comment]
I fixed the a/an. Can it be merged?

File: index.bs (hunk 1 above)
Comment (youennf) [reply index.bs]: Done

**Pull request merged**